### ISSUE TITLE
Parallel statement: compiler crash with examples from manual

### ISSUE THREAD

**Issue by mratsim** [nim-lang/Nim#11921]
I'm going through the example in the manual for the `parallel` statement but they don't use `spawn` which seems to be required, and using spawn crashes the compiler.

See https://nim-lang.org/docs/manual.html#pragmas-experimental-pragma

Invalid examples are:

```Nim
{.experimental: "parallel".}

proc useParallel() =
  parallel:
    for i in 0..4:
      echo "echo in parallel"
```

and

```Nim
# client.nim
proc useParallel*[T](unused: T) =
  # use a generic T here to show the problem.
  {.push experimental: "parallel".}
  parallel:
    for i in 0..4:
      echo "echo in parallel"
  
  {.pop.}
```

and the fixed one:

```Nim
import threadpool

{.experimental: "parallel".}

proc useParallel() =
  parallel:
    for i in 0..4:
      spawn echo("echo in parallel")

useParallel()
```

crashes with
```Nim
Error: internal error: (filename: "ccgexprs.nim", line: 1040, column: 17)
No stack traceback available
To create a stacktrace, rerun compilation with ./koch temp c <file>
```

**Comment by narimiran** [comment]
This seems like a duplicate of: https://github.com/nim-lang/Nim/issues/10913.

Also, see a minimal example at https://github.com/nim-lang/Nim/issues/9691.
